### ISSUE TITLE
VSCODE/Fluter Code completion not working for widget

### ISSUE THREAD

**Issue by ldsouza93** [Dart-Code/Dart-Code#4998]
when i start typing common dart code the completion is shown:

![image](https://github.com/Dart-Code/Dart-Code/assets/18901976/e2582fd1-c60c-4df1-aac0-3c68dd71ca81)

but when typing flutter widgets only the github copilot suggestion shows up

![image](https://github.com/Dart-Code/Dart-Code/assets/18901976/468fe020-6bfd-428e-be64-835c7bd3644d)

i tried reinstalling vscode, downgrading dart extension, rebooting pc, purging cache, but nothing helped.
i think the issue started after flutter 3.19 update.

with android studio everything works fine
![image](https://github.com/Dart-Code/Dart-Code/assets/18901976/316d5229-1ce6-439c-90d1-bf2e620a1c4e)
even with copilot
![image](https://github.com/Dart-Code/Dart-Code/assets/18901976/17b89ff1-27e7-4375-bdac-3352f10163f0)

 - Operating System and version: Windows 10
 - VS Code version: Version: 1.86.2 (user setup)
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:40:56.878Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.19045
 - Dart extension version: v3.82.0
 - Dart/Flutter SDK version: Flutter 3.19.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision bae5e49bc2 (7 days ago) • 2024-02-13 17:46:18 -0800
Engine • revision 04817c99c9
Tools • Dart 3.3.0 • DevTools 2.31.1

**Comment by DanTup** [comment]
In your first screenshot, you the code completion only seems to include snippets (`for`/`for in`). Can you confirm whether you're also seeing other code completion (such as class names), or is completion only providing snippets?

I'm not sure what the intended functionality is when using inline suggestions from Copilot. I found a few issues suggesting it does suppress normal completions, however it also seems like that was reverted, so I'm not sure if that might be related to what you're seeing here:

- https://github.com/microsoft/vscode/issues/165692
- https://github.com/microsoft/vscode/issues/190727
- https://github.com/microsoft/vscode/issues/185831
- https://github.com/microsoft/vscode/pull/190861

**Comment by ldsouza93** [comment]
> In your first screenshot, you the code completion only seems to include snippets (`for`/`for in`). Can you confirm whether you're also seeing other code completion (such as class names), or is completion only providing snippets?
> 
> I'm not sure what the intended functionality is when using inline suggestions from Copilot. I found a few issues suggesting it does suppress normal completions, however it also seems like that was reverted, so I'm not sure if that might be related to what you're seeing here:
> 
> * [Github copilot suggestion causes the system's own prompt not to be displayed microsoft/vscode#165692](https://github.com/microsoft/vscode/issues/165692)
> * [Stop supporting suppressSuggestions API microsoft/vscode#190727](https://github.com/microsoft/vscode/issues/190727)
> * [inline suppress suggestions doesn't work anymore microsoft/vscode#185831](https://github.com/microsoft/vscode/issues/185831)
> * [Fixes #190727 microsoft/vscode#190861](https://github.com/microsoft/vscode/pull/190861)

i guess i found the solution, i was getting slow suggestions a time ago and i came across the option ""dart.useLegacyAnalyzerProtocol": true," [in this post](https://stackoverflow.com/questions/65376213/vscode-flutter-dart-slow-suggestions) and been using it for better performance

but after commenting out this line and reloading vscode the widget suggestions returned

![image](https://github.com/Dart-Code/Dart-Code/assets/18901976/01e0c7bc-d6d1-46b1-a95e-ceb634c6561d)

the completion suggestions aren't too slow without this command but if there was anything to make it faster it'd be great too

my current settings.json is as 

```dart
{
    "notebook.showFoldingControls": "always",
    "files.autoSave": "afterDelay",
    "terminal.integrated.scrollback": 10000000,
    "git.enableSmartCommit": true,
    "git.openRepositoryInParentFolders": "never",
    "typescript.tsserver.maxTsServerMemory": 1024,
    "workbench.colorTheme": "One Dark Pro Monokai Darker",
    "editor.suggest.preview": true,
    "editor.minimap.enabled": false,
    "editor.showFoldingControls": "always",
    "editor.defaultFoldingRangeProvider": "Dart-Code.dart-code",
    "editor.foldingStrategy": "indentation",
    "editor.scrollBeyondLastColumn": 50,
    "dart.analysisServerFolding": false,
    // "dart.useLegacyAnalyzerProtocol": true,
    "dart.debugExternalPackageLibraries": true,
    "dart.debugSdkLibraries": true,
    "dart.openDevTools": "flutter",
    "dart.lineLength": 130,
    "dart.flutterRunAdditionalArgs": [
      "--verbose"
    ],
    "[dart]": {
      "editor.selectionHighlight": false,
      "editor.detectIndentation": false,
      "editor.formatOnSave": true,
      "editor.formatOnType": true,
      "editor.insertSpaces": true,
      "editor.tabCompletion": "onlySnippets",
      "editor.suggestSelection": "first",
      "editor.wordBasedSuggestions": "matchingDocuments",
      "editor.tabSize": 2,
      "editor.suggest.insertMode": "replace",
      "editor.defaultFormatter": "Dart-Code.dart-code",
      "editor.inlayHints.enabled": "offUnlessPressed"
    }
  }
```

**Comment by DanTup** [comment]
Ah yes, you've hit a problem I discovered yesterday (https://github.com/Dart-Code/Dart-Code/issues/5001). The completion support in the legacy protocol has changed and no longer works for the Dart-Code integration.

The next version of Dart-Code will ignore the `useLegacyProtocol` flag for new SDKs where it will not work correctly. This flag was not really intended to still be used so if you have issues using the new protocol, please do file issues as they need to be fixed.

I'll close this as a dupe of https://github.com/Dart-Code/Dart-Code/issues/5001
